runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -202,7 +202,16 @@ protected ILogData getNextEntry(QueuedStreamContext context,
      *
      * @param address       The address to read.
      */
-    protected abstract @Nonnull ILogData read(final long address);
+    protected @Nonnull ILogData read(final long address) {
+        try {
+            return readImpl(address);
```

Comment:
Ok I looked at the code now. The choices we have
1. create xxImpl style methods and wrap them with try catch .  
2. try catch down into backpointer stream view because no other implementation will use these abstractions.
3. move the read and readAll method implementations into the abstract class  and wrap them there.

At the moment the methods are so small, I do not have a big preferrence. If 3 is correct and makes sense to everybody we can do 3. else we can do 2.